dart/math/MathTypes.h

```diff
@@ -83,8 +85,28 @@ typedef Matrix<double, 6, 6> Matrix6d;
 typedef std::vector<Eigen::Vector3d> EIGEN_V_VEC3D;
 typedef std::vector<std::vector<Eigen::Vector3d > > EIGEN_VV_VEC3D;
 
+template <typename _Tp>
+using aligned_vector = std::vector<_Tp, Eigen::aligned_allocator<_Tp>>;
+
+template <typename _Key, typename _Tp, typename _Compare = std::less<_Key>>
+using aligned_map = std::map<_Key, _Tp, _Compare,
+    Eigen::aligned_allocator<std::pair<const _Key, _Tp>>>;
+
+template<typename _Tp, typename... _Args>
+inline std::shared_ptr<_Tp> make_aligned_shared()
```

Comment:
Yes, it's not for template recursion.

I found that `make_aligned_shared(...)` doesn't work for `WeldJoint::Property`; there was compiler error as:
```cpp
/usr/include/eigen3/Eigen/src/Core/util/Memory.h:731: error: call to implicitly-deleted copy constructor of 'std::_Sp_counted_ptr_inplace<dart::dynamics::WeldJoint::Properties, Eigen::aligned_allocator<dart::dynamics::WeldJoint::Properties>, __gnu_cxx::_Lock_policy::_S_atomic>'
        ::new( p ) T( value );
                   ^  ~~~~~
```
I put this argument-free overloaded function to make `make_aligned_shared(...)` looks like working with `WeldJoint::Property` temporally. Once the solution is found, this function should be removed.

It seems this error is present on gcc but not on clang. I've tried to figure out the reason of errors but couldn't find yet. Could you take a look at this problem?